### ISSUE TITLE
[DISCUSSION] Consider increasing default host memory limit per dask-cuda-worker

### ISSUE THREAD

**Issue by randerzander** [rapidsai/dask-cuda#169]
Several users have reported problems where `dask-cuda-worker` processes die in unexpected ways. After some debugging they find it's due to exceeding host memory limits, particularly when loading large training sets into GPU memory.

This is surprising for users, as it's not clear when or how a significant amount of host memory might be used, especially considering RAPIDS projects are focused on running as much as possible on GPUs.

**Comment by pentschev** [comment]
@randerzander we started the conversation about this offline, could you add a bit more context/examples of when things fail for you? Also, based on the experience you had, how did you setup the memory limit that generally worked?

cc @mrocklin for visibility

**Comment by mrocklin** [comment]
cc @quasiben for visibility

**Comment by pentschev** [comment]
@randerzander @beckernick @VibhuJawa is this still relevant? Is there some additional information you could share as to what would better defaults look like?

**Comment by jakirkham** [comment]
Friendly nudge @randerzander @beckernick @VibhuJawa 😉

**Comment by beckernick** [comment]
Thanks for the bump John. Anecdotally, we find that the most effective setup includes setting the host memory limit as the maximum available system memory (`(free -m | awk '/^Mem:/{print $2}')`). I'm interested to hear if folks thinks the system maximum is too high for a default.

**Comment by pentschev** [comment]
IMO, this would be too dangerous for a default. It seems that this was the best setup for TPCx-BB which was running in an exclusive environment, but this is not gonna be the case for every dask-cuda user, for instance running such a setup on a desktop being shared with other running applications may render the system very unstable due to main memory going completely full.

**Comment by beckernick** [comment]
> IMO, this would be too dangerous for a default. ...

I agree with Peter here. What's most effective for a given workflow doesn't necessarily translate to what's most effective for a default. A quick thought, though:

Naively, I'd expect Dask to start spilling at 60/70% host memory capacity, and then terminate at 95%. This feels to me like a good default for termination. We've made a lot of changes since last November. Is exceeding host memory while reading large files still as big of an issue? Is it possible this was related to spilling issues rather than host memory capacity issues?
